firmware/doc.txt

```diff
@@ -1,18 +1,46 @@
 /**
 @defgroup m4a-firmware  Reference firmware
 
-### What goes here?
-Specific documentation for our reference firmware ; m4a-firmware
+### Firmware
 
-You can find specific documentation for the sys, peripherals, and network folders.
+This is the core of Mesh4All ,You can find specific documentation for the sys, peripherals, and network folders.
 
+### Network
 
-### network
-UDP client, UDP server.
+This section has modules like:
 
-### peripherals
-Humidity and temperature sensors.
+    - UDP client: Make use of Internet Protocol of the TCP/IP suit.
 
-### sys
-AT commands, udp serialization, storage, ipv6 address.
+    - UDP server: Process all request sent by an udp client device.
+
+    - RPL protocol: The functions to start RPL and add a node dodag (Root)
+or dag (Slave).
+
+    - Radio: Module that handles all low power and low consumption wireless devices.
+
+    - Net tools: Contains all about networks tools.
+
+    - Chamos (Communication Handler for Addressing Management | Origin Server): It's used for
+server-side communication.
+
+    - Border Router: The module sets an IPV6 address in a wireless or wired interface.
+
+### Peripherals
+
+    - Moisture sensor: Gets the data of the moisture sensor.
```

Comment:
This is redundant, "I suggest something like : Moisture sensor: provides read humidity and temperature data of an enviroment ". Something like that